app.rb

```diff
@@ -3,20 +3,78 @@
 require_relative 'book'
 require_relative 'rental'
 require_relative 'manage_people'
-
+require 'json'
+# define class app
 class App
   attr_accessor :books, :people, :rentals
 
   def initialize
     @books = []
     @people = []
     @rentals = []
+    load_data
+  end
+
+  def save_books_data
+    file_data = @books.map { |book| { 'title' => book.title, 'author' => book.author } }
+    File.write('./data/books.json', JSON.generate(file_data))
+  end
+
+  def save_people_data
+    file_data = @people.map { |person| { 'name' => person.name, 'id' => person.id, 'age' => person.age } }
+    File.write('./data/people.json', JSON.generate(file_data))
+  end
+
+  def save_rentals_data
+    file_data = @rentals.map do |rental|
+      { 'date' => rental.date, 'book' => { 'title' => rental.book.title, 'author' => rental.book.author },
+        'person' => { 'name' => rental.person.name, 'id' => rental.person.id, 'age' => rental.person.age } }
+    end
+    File.write('./data/rentals.json', JSON.generate(file_data))
   end
 
   def list_books
     @books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
   end
 
+  def load_data
+    load_books_data
+    load_people_data
+    load_rentals_data
+  end
+
+  def load_books_data
+    file_data = File.read('./data/books.json')
+    @books = if file_data
+               JSON.parse(file_data).map do |book_data|
+                 Book.new(book_data['title'], book_data['author'])
+               end
+             else
+               []
+             end
+  end
+
+  def load_people_data
+    file_data = File.read('./data/people.json')
+    return unless file_data && !file_data.empty?
+
+    @people = JSON.parse(file_data).map do |person_data|
+      Person.new(person_data['age'], person_data['name'])
+    end
+  end
```

Comment:
When listing people data after I exit the application and re-run it again, instead of getting the type of each person, (Student or Teacher) I get [PERSON] displayed:

![image](https://github.com/0sugo/school_library/assets/93520178/fb4e3b26-9d0a-4f09-b183-29baa7e1a7c9)

Kindly fix this issue.